File: 2540. Minimum Common Value/2540.myans.py
class Solution:
    def getCommon(self, nums1: List[int], nums2: List[int]) -> int:
        len1 = len(nums1)
        len2 = len(nums2)

        idx1 = 0
        idx2 = 0

        while idx1 < len1 and idx2 < len2:
            n1 = nums1[idx1]
            n2 = nums2[idx2]

            if n1 == n2:
                return n1
            elif n1 < n2:
                idx1 += 1
            else:
                idx2 += 1
        
        return -1

File: 2540. Minimum Common Value/2540.myans.rs
impl Solution {
    pub fn get_common(nums1: Vec<i32>, nums2: Vec<i32>) -> i32 {
        let len1 = nums1.len();
        let len2 = nums2.len();

        let mut idx1 = 0usize;
        let mut idx2 = 0usize;

        while idx1 < len1 && idx2 < len2 {
            let n1 = nums1[idx1];
            let n2 = nums2[idx2];
            if n1 == n2 {
                return n1;
            } else if n1 < n2 {
                idx1 += 1;
            } else {
                idx2 += 1;
            }
        }

        -1
    }
}